starfish/experiment/experiment.py

```diff
@@ -81,6 +80,36 @@ def __repr__(self):
             f"    {images}"
         )
 
+    def parse_coordinate_groups(self, tileset: TileSet) -> List[CropParameters]:
+        """Takes a tileset and compares the physical coordinates on each tile to
+         create aligned coordinate groups (groups of tiles that have the same physical coordinates)
+
+         Returns
+         -------
+         A list of CropParameters. Each entry describes the r/ch/z values of tiles that are aligned
+         (have matching coordinates)
+         """
+        coord_groups: Dict[tuple, CropParameters] = dict()
+        for tile in tileset._tiles:
+            x_y_coords = tuple(i for i in [
+                tile.coordinates[Coordinates.X][0], tile.coordinates[Coordinates.X][1],
+                tile.coordinates[Coordinates.Y][0], tile.coordinates[Coordinates.Y][1],
+            ])
+            # A tile with this (x, y) has already been seen, add tile's Indices to CropParameters
+            if x_y_coords in coord_groups:
+                crop_params = coord_groups[x_y_coords]
+                crop_params.add_permitted_axes(Axes.CH, tile.indices[Axes.CH])
+                crop_params.add_permitted_axes(Axes.ROUND, tile.indices[Axes.ROUND])
+                if Axes.ZPLANE in tile.indices:
+                    crop_params.add_permitted_axes(Axes.ZPLANE, tile.indices[Axes.ZPLANE])
+            else:
+                coord_groups[x_y_coords] = CropParameters(
+                    permitted_chs=[tile.indices[Axes.CH]],
+                    permitted_rounds=[tile.indices[Axes.ROUND]],
+                    permitted_zplanes=[tile.indices[Axes.ZPLANE]] if Axes.ZPLANE in tile.indices
+                    else None)
+        return list(coord_groups.values())
```

Comment:
In python 3.6, dictionary iteration yielding the same order as insertion is an implementation detail, and not mandatory.  Therefore, multiple runs of this code can theoretically yield different orders.  I would consider sorting the keys, and using that to return a list of CropParameters.

In python 3.7, that has become a language level edict, so it can be relied on.  Unfortunately, we are on python 3.6.  But you can drop a comment in here saying that the complexity can be removed when we move to 3.7. :)

Reply:
Changed to use a OrderedDict